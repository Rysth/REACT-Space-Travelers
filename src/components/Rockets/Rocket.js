import PropTypes from 'prop-types';

export default function Rocket({
  photo, descripte, name,
}) {
  return (
    <section>
      <div>
        <div>
          <img src={photo} alt={name} />
        </div>
        <div className="col-9">
          <h3>{name}</h3>
          <p>{descripte}</p>
          <button >Reserve</button>
        </div>
      </div>
    </section>
  );
}

Rocket.propTypes = {
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};
