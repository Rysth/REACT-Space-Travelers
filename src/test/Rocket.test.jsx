import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import store from '../redux/Store';
import Rocket from '../components/Rockets/Rocket';

describe('Test', () => {
  const rocket = (
    <Provider store={store}>
      <Rocket
        description="Rockets"
        id="1"
        image="none"
        isReserved={false}
        name="roket"
      />
    </Provider>
  );

  test('Create snapshot for rocket', () => {
    const component = render(rocket);
    expect(component).toMatchSnapshot();
  });

  test('Rocket has a button element', () => {
    render(rocket);
    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();
  });

  test('Rocket has a description', () => {
    render(rocket);
    const description = screen.getByText('Rockets');
    expect(description).toBeInTheDocument();
  });

  test('It has an image component', () => {
    render(rocket);
    const image = screen.getByRole('img');
    expect(image).toBeInTheDocument();
  });

  test('It has a button with text Reserve Rocket', () => {
    render(rocket);
    const button = screen.getByRole('button');
    expect(button).toHaveTextContent('Reserve Rocket');
  });

  test('Rocket has a name', () => {
    render(rocket);
    const name = screen.getByText('roket');
    expect(name).toBeInTheDocument();
  });
});
