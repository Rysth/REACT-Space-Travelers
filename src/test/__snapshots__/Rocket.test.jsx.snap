// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rocket component Create snapshot for rocket 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="flex justify-start px-16"
        >
          <div
            class="w-52"
          >
            <img
              alt="roket"
              class="img"
            />
          </div>
          <div
            class="w-3/4 ml-4 text-base"
          >
            <h3
              class="font-bold text-lg mt-2"
            >
              roket
            </h3>
            <p>
              Rockets
            </p>
            <button
              class="bg-blue-600 px-4 py-1 text-white rounded mt-2"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="flex justify-start px-16"
      >
        <div
          class="w-52"
        >
          <img
            alt="roket"
            class="img"
          />
        </div>
        <div
          class="w-3/4 ml-4 text-base"
        >
          <h3
            class="font-bold text-lg mt-2"
          >
            roket
          </h3>
          <p>
            Rockets
          </p>
          <button
            class="bg-blue-600 px-4 py-1 text-white rounded mt-2"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
